generate_openj9_scc.sh

```diff
@@ -0,0 +1,295 @@
+#!/bin/bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+unset OPENJ9_JAVA_OPTIONS
+
+# Default size for SCC
+SCC_SIZE="50m"
+
+# Runs for generating SCC
+SCC_GEN_RUNS_COUNT=3
+
+# is running on ubuntu or debian - package manager apt
+IS_APT_ENV=false
+# Intitialise default / allowed sample app status
+RUN_ECLIPSE=false
+RUN_TOMCAT=true
+
+# Intitialise default / required packages status
+INSTALL_GTK=false
+INSTALL_XVFB=false
+
+# App download locations
+DOWNLOAD_PATH_ECLIPSE=/tmp/eclipse
+DOWNLOAD_PATH_TOMCAT=/tmp/tomcat
+
+# App installation locations
+INSTALL_PATH_ECLIPSE="$HOME"/eclipse-home
+INSTALL_PATH_TOMCAT="$HOME"/tomcat-home
+
+# function to install packages
+function install_packages() {
+    if [ "$IS_APT_ENV" == true ]; then
+        # update the repositories
+        apt update
+
+        # Set non-interactive frontend (to avoid `tzdata` config options selection)
+        APT_INSTALL_CMD="DEBIAN_FRONTEND=noninteractive"
+        APT_INSTALL_CMD="${APT_INSTALL_CMD} apt install -y"
+
+        if [ "$INSTALL_GTK" == true ]; then
+            # install gtk+3.0
+            APT_INSTALL_CMD="${APT_INSTALL_CMD} gtk+3.0"
+        fi
+
+        if [ "$INSTALL_XVFB" == true ]; then
+            # install xvfb
+            APT_INSTALL_CMD="${APT_INSTALL_CMD} xvfb"
+        fi
+
+        eval "$APT_INSTALL_CMD"
+    fi
+}
+
+# function to download the applications
+function download_and_install_artifacts() {
+
+    # check for eclipse
+    if [ "$RUN_ECLIPSE" == true ]; then
```

Comment:
Each of these needs to have a function on its own. It is a lot easier to parse and understand that way (including the `if` check)

Reply:
Created seperate functions. Thanks for the review